core/pom.xml

```diff
@@ -46,5 +52,15 @@
                 <directory>src/main/resources</directory>
             </resource>
         </resources>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <version>3.7.0</version>
+                <configuration>
+                    <sourcepath>src/main/java;target/generated-sources/annotations</sourcepath>
```

Comment:
Please check that the source plugin also contains the generated source code of the class :)